compiler/rustc_mir_build/src/build/expr/as_constant.rs

```diff
@@ -27,7 +27,12 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 Constant { span, user_ty, literal: literal.into() }
             }
             ExprKind::StaticRef { literal, .. } => {
```

Comment:
We could make `StaticRef` only contain an allocid and a type and build the mir constant from scratch here.

I think at the creation site of StaticRef we do this work only to partially revert it here